modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts

```diff
@@ -32,6 +34,46 @@ import { KeychainsTriplet } from '../../../baseCoin';
  */
 
 export class EddsaUtils extends baseTSSUtils<KeyShare> {
+  async verifyWalletSignatures(
+    userGpgKey: openpgp.SerializedKeyPair<string>,
+    bitgoKeychain: Keychain,
+    decryptedShare: string,
+    verifierIndex: 1 | 2
+  ): Promise<void> {
+    assert(bitgoKeychain.commonKeychain);
+    assert(bitgoKeychain.walletHSMGPGPublicKeySigs);
+
+    const bitgoGpgKey = await getBitgoGpgPubKey(this.bitgo);
+
+    const userKeyPub = await openpgp.readKey({ armoredKey: userGpgKey.publicKey });
+    const userKeyId = userKeyPub.keyPacket.getFingerprint().padStart(40, '0');
+
+    const walletSignatures = await openpgp.readKeys({ armoredKeys: bitgoKeychain.walletHSMGPGPublicKeySigs });
+    if (walletSignatures.length !== 2) {
+      throw new Error('Invalid wallet signatures');
+    }
+
+    await verifyWalletSignature({
+      walletSignature: walletSignatures[0],
+      commonKeychain: bitgoKeychain.commonKeychain,
+      userKeyId,
+      backupKeyId: userKeyId,
```

Comment:
why are you using userKeyId for the backupKeyId?

Reply:
Because we haven't built anything yet that allows a third party to store the backup key shares. So far the user always  creates and stores the backup key shares and reuses the user gpg key as the backup gpg key.
Once we actually implement code to allow for a third party KRS we can change this code.

Reply:
Excellent point. I still think that we currently have some valid use cases where the user and backup gpg key are the same but I updated this function so that they are separate.